spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/procedures/SetCurrentSnapshotProcedure.java

```diff
@@ -78,17 +83,21 @@ public StructType outputType() {
   @Override
   public InternalRow[] call(InternalRow args) {
     Identifier tableIdent = toIdentifier(args.getString(0), PARAMETERS[0].name());
-    long snapshotId = args.getLong(1);
+    Long snapshotId = args.isNullAt(1) ? null : args.getLong(1);
+    String ref = args.isNullAt(2) ? null : args.getString(2);
+    Preconditions.checkArgument(
+        snapshotId != null || ref != null, "snapshot_id and ref cannot both be null");
```

Comment:
I'd expect this to throw an exception if both are provided.
What about something like below?

```
Preconditions.checkArgument(
    (snapshotId != null && ref == null) || (snapshotId == null && ref != null),
    "Either snapshot_id or ref must be provided, not both");
```